charts/gardener/controlplane/charts/runtime/templates/admission-controller/deployment.yaml

```diff
@@ -0,0 +1,107 @@
+  labels:
+    app: gardener
+    role: admission-controller
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    release: "{{ .Release.Name }}"
+    heritage: "{{ .Release.Service }}"
+spec:
+  replicas: {{ required ".Values.global.admission.replicaCount is required" .Values.global.admission.replicaCount }}
+  selector:
+    matchLabels:
+      app: gardener
+      role: admission-controller
+  template:
+    metadata:
+      annotations:
+        checksum/configmap-gardener-admission-controller-config: {{ include (print $.Template.BasePath "/admission-controller/configmap-componentconfig.yaml") . | sha256sum }}
+        checksum/secret-admission-controller-cert: {{ include (print $.Template.BasePath "/admission-controller/secret-cert.yaml") . | sha256sum }}
+        checksum/secret-gardener-admission-controller-kubeconfig: {{ include (print $.Template.BasePath "/admission-controller/secret-kubeconfig.yaml") . | sha256sum }}
+        {{- if .Values.global.admission.podAnnotations }}
+{{ toYaml .Values.global.admission.podAnnotations | indent 8 }}
+        {{- end }}
+      labels:
+        app: gardener
+        role: admission-controller
+        chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+        release: "{{ .Release.Name }}"
+        heritage: "{{ .Release.Service }}"
+        {{- if .Values.global.admission.podLabels }}
+{{ toYaml .Values.global.admission.podLabels | indent 8 }}
+        {{- end }}
+    spec:
+      {{- if not .Values.global.admission.kubeconfig }}
+      serviceAccountName: {{ required ".Values.global.admission.serviceAccountName is required" .Values.global.admission.serviceAccountName }}
+      {{- end }}
+      {{- if gt (int .Values.global.admission.replicaCount) 1 }}
+      affinity:
+        podAntiAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+          - labelSelector:
+              matchExpressions:
+              - key: app
+                operator: In
+                values:
+                - gardener
+              - key: role
+                operator: In
+                values:
+                - admission-controller
+            topologyKey: kubernetes.io/hostname
+      {{- end }}
+      containers:
+      - name: gardener-admission-controller
+        image: {{ include "utils-templates.image" .Values.global.admission.image }}
+        imagePullPolicy: {{ .Values.global.admission.image.pullPolicy }}
+        command:
+        - /gardener-admission-controller
+        - --config=/etc/gardener-admission-controller/config/config.yaml
+        {{- if .Values.global.admission.resources }}
+        resources:
+{{ toYaml .Values.global.admission.resources | indent 10 }}
+        {{- end }}
+        livenessProbe:
+          httpGet:
+            path: /healthz
+            port: {{ required ".Values.global.admission.config.server.https.port is required" .Values.global.admission.config.server.https.port }}
+            scheme: HTTPS
+          initialDelaySeconds: 300
```

Comment:
Any reason for 5m initial delay?

Reply:
Yes, it's the timeout for the cache wait sync. Otherwise we might already contact the `healthz` endpoint even though the initialization hasn't finished yet.